fetching/directory-fetcher/src/index.ts

```diff
@@ -129,15 +128,35 @@ async function fetchPackageFilesFromDir (
   const files = await packlist({ path: dir })
   const filesIndex: Record<string, string> = Object.fromEntries(files.map((file) => [file, path.join(dir, file)]))
   if (opts.manifest) {
-    // In a regular pnpm workspace it will probably never happen that a dependency has no package.json file.
-    // Safe read was added to support the Bit workspace in which the components have no package.json files.
-    // Related PR in Bit: https://github.com/teambit/bit/pull/5251
-    const manifest = await safeReadProjectManifestOnly(dir) ?? {}
-    opts.manifest.resolve(manifest as any) // eslint-disable-line @typescript-eslint/no-explicit-any
+    const manifest = await safeReadProjectManifestAndMakeExportable(dir, filesIndex)
+    opts.manifest.resolve(manifest ?? {} as any) // eslint-disable-line @typescript-eslint/no-explicit-any
   }
   return {
     local: true as const,
     filesIndex,
     packageImportMethod: 'hardlink' as const,
   }
 }
+
+async function safeReadProjectManifestAndMakeExportable (
+  dir: string,
+  filesIndex: Record<string, string>
+): Promise<ProjectManifest | null> {
+  // In a regular pnpm workspace it will probably never happen that a dependency has no package.json file.
+  // Safe read was added to support the Bit workspace in which the components have no package.json files.
+  // Related PR in Bit: https://github.com/teambit/bit/pull/5251
+  const manifest = await safeReadProjectManifest(dir)
+
+  if (manifest) {
+    const { fileName } = manifest
+    const exportableManifest = await createExportableManifest(
+      dir,
+      manifest.manifest
+    )
+    const manifestPathOverride = path.join(dir, 'node_modules', '.pnpm', fileName)
+    await manifest.writeProjectManifest(exportableManifest, { manifestPathOverride })
```

Comment:
Here is the use case of `manifestPathOverride`. As mentioned in other comment, could replace this call with raw `writeProjectManifest` (rather than the one built into the safeRead response)